python/lancedb/rerankers/cohere.py

```diff
@@ -0,0 +1,51 @@
+from typing import Union
+from functools import cached_property
+from .base import Reranker
+import lancedb
+import pyarrow as pa
+import os
+
+
+class CohereReranker(Reranker):
+    """
+    Reranks the results using cohere rerank api.
+
+    Parameters
+    ----------
+    model_name : str, default "rerank-multilingual-v2.0"
+        The name of the cross encoder model to use. Available cohere models are:
+        - rerank-english-v2.0
+        - rerank-multilingual-v2.0
+    column : str, default "text"
+        The name of the column to use as input to the cross encoder model.
+    top_k : str, default None
+        The number of results to return. If None, will return all results.
+    """
+    def __init__(self, model_name:str = "rerank-multilingual-v2.0", column:str = "text", top_n: Union[int, None]=None):
+        self.model_name = model_name
+        self.column = column
+        self.top_n = top_n
+        
+    @cached_property
+    def client(self):
+        cohere = self.safe_import("cohere")
+        if os.environ.get("COHERE_API_KEY") is None:
+                self.api_key_not_found_help("cohere")
+        return cohere.Client(os.environ["COHERE_API_KEY"])
+    
+    def rerank_hybrid(self, query_builder: "lancedb.HybridQueryBuilder" , vector_results: pa.Table, fts_results: pa.Table, combined_results: pa.Table):
```

Comment:
why do we still need the `vector_results` and `fts_results`?

Reply:
Users might want to use their custom logic to combine those results.

Reply:
@AyushExel I presume there will be a guides page in the docs on how users can do that? It's not immediately obvious that the user has the option to use their custom logic.

Reply:
yeah will document. Just changed the api a bit too